### PR TITLE
Fixes undefined vars in some migrated controllers

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Used phpstan static code analyzer locally to collect errors from Domain, Core and PrestashopBundle/Controllers of migrated pages. Found some possibly undefined vars in some of controllers. Removed possibility to try rendering a page with undefined variable.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Approve that following forms create/edit are rendering fine: `Catalog->Brand & Supplier`, `sell/catalog/catalog-price-rules `, `Catalog->Files`, `CustomerService->Order Message`. No changes visible.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17292)

### CODE REVIEW

**Comment by matks** [comment]
PrettyCI is not happy 😅

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php
Comment (matks) [inline src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php:166]: It looks like a good idea but I dont understand why you remove the previous statement ?
```
if ($e instanceof AttachmentNotFoundException || $e instanceof ErrorException) {
...
```

Comment (zuk3975) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php:166]: Phpstan wasnt passing this way. I figured, all we care is that those vars must be set and we dont need to check which exception fault is that. There were some controllers missing those catches, so i would need to dive into handlers and check which exception to catch. Seemed fast and convienient. Not sure if its good approach tho :D

Comment (matks) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php:166]: What I'm worried about is that some `ErrorException` is thrown from `$result = $attachmentFormHandler->handleFor((int) $attachmentId, $attachmentForm);` BUT `attachmentForm` and `attachmentInformation` are set

So please put back the exception 😉

Comment (zuk3975) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php:166]: @RaimondasSapola, why you used this ErrorException, was it by mistake? I don't see it being thrown or used anywhere else.

Comment (RaimondasSapola) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php:166]: @zuk3975  If i remember correctly this exception can be thrown when deleting old file that doesn't exist. In legacy code when unlinking file warnings were suppressed and with Attachment Forms PR a possibility to throw an exception was introduced. So what was done is i just removed warning suppression and symfony handles it and throws an ErrorException.

Comment (zuk3975) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php:166]: According to https://github.com/PrestaShop/PrestaShop/pull/15003#discussion_r323268073, I've came up with solution. Catch ErrorException as fast as possilbe and throw CannotUnlinkAttachmentException, by which I know in controller for sure why and when it happened, so i can add warning flash instead of error (because the edit action actually was successful). @matks wdyt?

Comment (matks) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php:166]: @zuk3975 I love this solution 👍 !

Comment (zuk3975) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php:166]: so its done :smile:

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
I approve this PR, waiting for another approval before QA

File: src/Adapter/File/Uploader/AttachmentFileUploader.php
Comment (matthieu-rolland) [inline src/Adapter/File/Uploader/AttachmentFileUploader.php]: It was so much easier to read when it was not on one line, is there a reason?

Comment (zuk3975) [reply src/Adapter/File/Uploader/AttachmentFileUploader.php]: no, can it be cs fixer? Ill check

Comment (zuk3975) [reply src/Adapter/File/Uploader/AttachmentFileUploader.php]: Yup i checked, cs fixer puts it in one line :octocat:

Comment (matthieu-rolland) [reply src/Adapter/File/Uploader/AttachmentFileUploader.php]: oh ok...  👍

Comment (zuk3975) [reply src/Adapter/File/Uploader/AttachmentFileUploader.php]: Its strange anyway, because I've pushed the multiline code without fixing and prettyCI passes.

**Review by matks — DISMISSED** [review]

File: src/Adapter/File/Uploader/AttachmentFileUploader.php
Comment (jolelievre) [reply src/Adapter/File/Uploader/AttachmentFileUploader.php]: Yeah it's a pain this new modif, I wonder if it's not related to `printf` rather than the exception

**Review by PierreRambaud — DISMISSED** [review]

**Comment by matks** [comment]
Unfortunately we now need a rebase 😢 before QA

**Comment by zuk3975** [comment]
> Unfortunately we now need a rebase  before QA

Applied cs fixer which put MILLION WORDS exception to one line and left no conflict. cs fixer should be configured at some point tho, it shouldn't create such unreadable one-liners.

**Comment by matks** [comment]
> > Unfortunately we now need a rebase  before QA
> 
> Applied cs fixer which put MILLION WORDS exception to one line and left no conflict. cs fixer should be configured at some point tho, it shouldn't create such unreadable one-liners.

It's already configured, however, the configuration might be improved 🤔 ?

**Review by matks — DISMISSED** [review]

File: src/Adapter/File/Uploader/AttachmentFileUploader.php
Comment (eternoendless) [inline src/Adapter/File/Uploader/AttachmentFileUploader.php]: This should be broken up in several lines.

File: src/Adapter/File/Uploader/AttachmentFileUploader.php
Comment (eternoendless) [reply src/Adapter/File/Uploader/AttachmentFileUploader.php]: I don't agree with this change. If cs fixer is doing it then it's misconfigured because it violates PSR-2 line length for no reason.

**Review by matks — DISMISSED** [review]

File: src/Adapter/File/Uploader/AttachmentFileUploader.php
Comment (matks) [reply src/Adapter/File/Uploader/AttachmentFileUploader.php]: Fixed

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/ManufacturerController.php
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/Sell/Catalog/ManufacturerController.php]:
```suggestion
            /** @var EditableManufacturerAddress $editableAddress */
```

**Review by jolelievre — COMMENTED** [review]
Is it possible to quickly fix this?

File: src/Adapter/File/Uploader/AttachmentFileUploader.php
Comment (jolelievre) [inline src/Adapter/File/Uploader/AttachmentFileUploader.php]: It would make more sense (and it would avoid "two" calls to unlink)to manage the exception throwing in the catch condition:
```php
try {
    unlink($fileLink);
} catch (ErrorException $e) {
    if ($throwExceptionOnFailure) {
        throw new CannotUnlinkAttachmentException($e->getMessage(), 0, null, $fileLink);
    }
}
```

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
Waiting for another review, to me code is good

**Review by PierreRambaud — DISMISSED** [review]

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php
Comment (rokaszygmantas) [inline src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php]: while these comments look useless, we still should keep them as agreed with @matks

File: src/Core/Domain/Attachment/Exception/CannotUnlinkAttachmentException.php
Comment (rokaszygmantas) [inline src/Core/Domain/Attachment/Exception/CannotUnlinkAttachmentException.php]: old licence

File: src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/OrderMessageController.php
Comment (rokaszygmantas) [inline src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/OrderMessageController.php:183]: same regarding redirect as above

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/SupplierController.php
Comment (rokaszygmantas) [inline src/PrestaShopBundle/Controller/Admin/Sell/Catalog/SupplierController.php:350]: maybe this redirect could be moved to `catch` block above? then you wouldn't need the `if` condition

**Review by rokaszygmantas — CHANGES_REQUESTED** [review]

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/SupplierController.php
Comment (zuk3975) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/SupplierController.php:350]: We want to render the form if its possible even if exception is thrown.
If i move redirect to catch and some random exception is thrown it will redirect to index instead of rendering the form. Although I wouldn't mind that, but I guess its ux based decision.

File: src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/OrderMessageController.php
Comment (zuk3975) [reply src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/OrderMessageController.php:183]: same answer as above :trollface:

**Review by rokaszygmantas — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by boubkerbribri** [comment]
I executed functional automatic tests on Catalog->Brand & Supplier, Catalog->Files, CustomerService->Order Message and it looks good.
@Robin-Fischer-PS can you finish the test please.

**Comment by Progi1984** [comment]
Thanks @zuk3975

**Pull request merged**
